be/src/olap/delta_writer.cpp

```diff
@@ -410,9 +417,14 @@ Status DeltaWriter::cancel() {
         _flush_token->cancel();
     }
     _is_cancelled = true;
+    _cancel_status = st;
     return Status::OK();
 }
 
+Status DeltaWriter::cancel() {
+    return cancel_with_status(Status::OK());
```

Comment:
Here, cancel_with_status(Status::OLAPInternalError(OLAP_ERR_ALREADY_CANCELLED)); 

 Then c

if (_is_cancelled) {
        RETURN_NOT_OK(_cancel_status);
        return Status::OLAPInternalError(OLAP_ERR_ALREADY_CANCELLED);
    }

can be written like below:
if (_is_cancelled) {
        return _cancel_status;
}.

Reply:
OLAPInternalError will generate stacktrace information. If the above method is used, the stacktrace information will be generated in advance.

Reply:
I think stacktrace information seems useless, so I modified it according to your suggestion.